teamworkApp/lib/dbCalls.py

```diff
@@ -0,0 +1,109 @@
+# muddersOnRails()
+# Sara McAllister November 5, 2017
+# Last updated: 11-8-2017
+
+# library for SQLite database calls for teamwork analysis app
+
+import contextlib
+import datetime
+import sqlite3
+
+DB = 'db/development.sqlite3'
+test_DB = 'db/test.sqlite3'
+
+def connect(sqlite_file):
+    """ Make connection to an SQLite database file """
+    conn = sqlite3.connect(sqlite_file)
+    c = conn.cursor()
+    return conn, c
+
+def close(conn):
+    """ Commit changes and close connection to the database """
+    conn.commit()
+    conn.close()
+
+@contextlib.contextmanager
+def dbconnect(test):
+    """Create cursor and ensure that DB connection closes"""
+    sqlite_file = DB
+    if test: sqlite_file = test_DB
+
+    conn, cursor = connect(sqlite_file)
+    try:
+        yield cursor
+    finally:
+        close(conn)
+
+def get_all_styles(test=False):
+    """Return all style entries in db ordered based on entry in db"""
+    with dbconnect(test) as c:
+        scores = c.execute('SELECT * FROM styles').fetchall()
+    return scores
+
+def get_students_styles(student_ids, test=False):
+    """Return style associated with a student's id"""
+    placeholders = ', '.join('?' for id in student_ids)
+    sql = 'SELECT * FROM styles WHERE student_id IN (%s)' % placeholders
+    with dbconnect(test) as c:
+        styles = c.execute(sql, student_ids).fetchall()
+    return styles
+
+def get_student_answers(student_id, test=False):
+    """Return all answers associated with a student's id"""
+    with dbconnect(test) as c:
+        answers = c.execute(
+            'SELECT * FROM answers WHERE student_id=? ORDER BY id DESC', 
+            [student_id,],
+        ).fetchall()
+    return answers
+
+def get_all_students(test=False):
+    """Return all students' information"""
+    with dbconnect(test) as c:
+        students = c.execute('SELECT * FROM students').fetchall()
+    return students
+
+def get_all_student_IDs(test=False):
+    """Return all students' information"""
+    with dbconnect(test) as c:
+        students = c.execute('SELECT id FROM students').fetchall()
+    return students
+
+def insert_student_team_pairs(student_and_team_ids, test=False):
```

Comment:
how is this list structured? Is it a LoL, as in [[studentID, teamID], [studentID, teamID]] or is it just pairwise connected, i.e. [student1ID, team1ID, student2ID, team2ID]? Can you clarify this in your docstring?